Update d log 21

YanZhu2022 opened the pull request:
# Changes

*This branch serves as the foundation for further enhancements, including incorporating format changes for the New D-log data (2024) and adding functionality to identify large sites

- Updated land_use_input path validation: Changed Optional[pydantic.FilePath] to Optional[pathlib.Path]. Unlike pydantic.FilePath, which immediately validates file existence when parsing the YAML, pathlib.Path (or str) allows validation to occur within a custom validator, ensuring run_land_use is properly checked before verifying the file path.
- Expanded Land_Use module: Added functions to convert site-level development data into dwellings and jobs.
- Minor fixes and improvements, such as update @pydantic.root_validator to @model_validator, along with other small fixes

Commit:
Added inputs and functions related to new zones based on LSOA

Changes related to site to new zones conversions are introduced including paths in yaml file for new zones, config paths in inputs, and major syntax changes in Land_use (addtional functions) to allow conversion from sites to zones

YanZhu2022 committed:
Minor fixes

Fixed bug of validating land_use_input path and other small fixes. Should have no issue to run with D-log 2021 data to produce site level dwelling and jobs.

CDuncan-TfN commented on d_lit-config_ss.yml:
Does this file need to exist?

YanZhu2022 replied:
nope. sorry that I forgot to remove it

CDuncan-TfN committed:
Remove duplicate functions

CDuncan-TfN committed:
Parameter description

CDuncan-TfN committed:
Remove redundant functions

More functions which duplicate existing functionality (minor modification to generalise original functions)

CDuncan-TfN committed:
Delete d_lit-config_ss.yml
